requests/models.py

```diff
@@ -813,7 +813,10 @@ def iter_lines(self, chunk_size=ITER_CHUNK_SIZE, decode_unicode=None, delimiter=
             #
             # If we're using `splitlines()`, we only do this if the chunk
             # ended midway through a line.
-            incomplete_line = lines[-1] and lines[-1].endswith(chunk[-1])
+            if isinstance(chunk[-1], basestring):  # Decoded string (decode_unicode=True or Py2)
+                incomplete_line = lines[-1] and lines[-1].endswith(chunk[-1])
+            else:  # Bytestring (decode_unicode=False and Py3)
+                incomplete_line = lines[-1] and lines[-1].endswith(chr(chunk[-1]).encode('utf-8'))
```

Comment:
Thanks, I missed that... much neater. It works fine with a simple indexing in all cases.